uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/utils/UnitFormatUtils.kt

```diff
@@ -0,0 +1,114 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.common.utils
+
+import com.sygic.maps.uikit.viewmodels.common.regional.units.DistanceUnit
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+import java.util.*
+import kotlin.math.roundToInt
+
+object Distance {
+
+    private const val METRIC_KM_DISTANCE_UNIT = "km"
+    private const val METRIC_M_DISTANCE_UNIT = "m"
+    private const val IMPERIALS_MI_DISTANCE_UNIT = "mi"
+    private const val IMPERIALS_FEET_DISTANCE_UNIT = "ft"
+    private const val IMPERIALS_YARD_DISTANCE_UNIT = "yd"
+    private const val M_TO_KM_CONVERSION_RATIO = 0.001f
+    private const val M_TO_MI_CONVERSION_RATIO = 0.00062f
+    private const val M_TO_FT_CONVERSION_RATIO = 3.28084f
+    private const val M_TO_YD_CONVERSION_RATIO = 1.09361f
+
+    private val DECIMAL_FORMAT = DecimalFormat("#.#", DecimalFormatSymbols(Locale.ENGLISH))
+
+    fun getFormattedDistance(
+        distanceUnit: DistanceUnit,
+        distanceInMeters: Int,
+        roundSmallUnits: Boolean = true
+    ): String = when (distanceUnit) {
+        DistanceUnit.KILOMETERS -> {
+            val kilometers = distanceInMeters * M_TO_KM_CONVERSION_RATIO
+            when {
+                kilometers >= 10 -> kilometers.roundToInt().toString() + METRIC_KM_DISTANCE_UNIT
+                kilometers >= 1 -> DECIMAL_FORMAT.format(kilometers.toDouble()) + METRIC_KM_DISTANCE_UNIT
+                roundSmallUnits -> getFormattedDistance(distanceUnit, roundDistance(distanceInMeters), false)
+                else -> distanceInMeters.toString() + METRIC_M_DISTANCE_UNIT
+            }
+        }
+
+        DistanceUnit.MILES_YARDS -> getDistanceInMilesWithUnits(
+            distanceInMeters,
+            M_TO_YD_CONVERSION_RATIO,
+            IMPERIALS_YARD_DISTANCE_UNIT,
+            roundSmallUnits
+        )
+
+        DistanceUnit.MILES_FEETS -> getDistanceInMilesWithUnits(
+            distanceInMeters,
+            M_TO_FT_CONVERSION_RATIO,
+            IMPERIALS_FEET_DISTANCE_UNIT,
+            roundSmallUnits
+        )
+    }
+
+    private fun getDistanceInMilesWithUnits(
+        meters: Int,
+        smallUnitConversionRatio: Float,
+        smallUnitText: String,
+        roundSmallUnits: Boolean
+    ): String {
+        val miles = meters * M_TO_MI_CONVERSION_RATIO
+        val smallUnitsValue = (meters * smallUnitConversionRatio).roundToInt()
+
+        return when {
+            miles >= 10 -> miles.roundToInt().toString() + IMPERIALS_MI_DISTANCE_UNIT
+            smallUnitsValue >= 1000 -> {
+                val roundedMiles = miles.toInt()
+                val rest = miles % 1
```

Comment:
tu ked clovek nevie, ze rest je float, alebo double, tak to na prvy pohlad nedava zmysel lebo cokolvek `%` 1 == 0

Pozeral som, ci uz kotlin nema nieco ako Float.getDecimal, ale asi nema. Mozno by to stalo za extension a tu by potom bolo iba `miles.getDecimal` alebo `miles.decimal`